### PR TITLE
enh: make viz responsive, remove extra renders, handle err in main window

### PR DESCRIPTION
**Pull request by fontanierh**
## Description

1. improve the responsiveness of viz components by changing the prompt
2. improve overall latency by not relying on onLoad (and instead relying on contentHeight)
3. stop displaying error through the iframe (instead, use a `setErrored` cross-doc call to set error state on parent window)
4. improve latency / smoothness by removing state vars (enabled by removing `onLoad` and not relying on iframe to display the error)

After this PR, we can release viz

## Risk

N/A

### CODE REVIEW

File: front/components/assistant/conversation/actions/VisualizationActionIframe.tsx [hunk 1]
```diff
@@ -130,45 +130,41 @@ export function VisualizationActionIframe({
...
   owner: WorkspaceType;
   visualization: Visualization;
-
   onRetry: () => void;
 }) {
-  const [contentHeight, setContentHeight] = useState(0);
-  const [iframeLoaded, setIframeLoaded] = useState(false);
-  const [showSpinner, setShowSpinner] = useState(true);
+  const [contentHeight, setContentHeight] = useState<number>(0);
+  const [errored, setErrored] = useState(false);
```
Comment (flvndvd) [inline front/components/assistant/conversation/actions/VisualizationActionIframe.tsx:136]: If you keep a boolean, let keep the same convention:
```suggestion
  const [isErrored, setIsErrored] = useState(false);
```

File: front/components/assistant/conversation/actions/VisualizationActionIframe.tsx [hunk 2]
```diff
@@ -182,9 +178,12 @@ export function VisualizationActionIframe({
         ? `${codeRef.current?.scrollHeight}px`
         : "100%";
     } else if (activeIndex === 1) {
-      containerRef.current.style.height = `${contentHeight}px`;
+      containerRef.current.style.height = !errored ? `${contentHeight}px` : "";
+      if (errored && errorRef.current) {
+        containerRef.current.style.height = `${errorRef.current.scrollHeight}px`;
+      }
```
Comment (flvndvd) [inline front/components/assistant/conversation/actions/VisualizationActionIframe.tsx:184]: `nit`
```suggestion
  let height = '';

  if (errored && errorRef.current) {
    height = `${errorRef.current.scrollHeight}px`;
  } else if (!errored) {
    height = `${contentHeight}px`;
  }

  containerRef.current.style.height = height;
```

File: front/components/assistant/conversation/actions/VisualizationActionIframe.tsx [hunk 3]
```diff
@@ -202,7 +201,9 @@ export function VisualizationActionIframe({
       <div
         className={classNames(
           "transition-height relative w-full overflow-hidden duration-500 ease-in-out",
-          codeFullyGenerated ? "min-h-96" : ""
+          codeFullyGenerated && !errored ? "min-h-96" : "",
```
Comment (flvndvd) [inline front/components/assistant/conversation/actions/VisualizationActionIframe.tsx:204]: 🤔 Don't we want the min height only when we have an error?

File: front/components/assistant/conversation/actions/VisualizationActionIframe.tsx [hunk 4]
```diff
@@ -219,21 +220,40 @@ export function VisualizationActionIframe({
...
           </div>
           <div className="relative flex h-full w-full shrink-0 items-center justify-center">
-            {codeFullyGenerated && (
+            {codeFullyGenerated && !errored && (
               <div
-                style={{ height: `${contentHeight}px` }}
+                style={{
+                  height: !errored ? `${contentHeight}px` : "100%",
+                  minHeight: !errored ? "96" : undefined,
+                  maxHeight: "60vh",
```
Comment (flvndvd) [inline front/components/assistant/conversation/actions/VisualizationActionIframe.tsx:228]: ✂️

File: viz/app/components/ErrorBoundary.tsx [hunk 5]
```diff
@@ -28,65 +26,14 @@ export class ErrorBoundary extends React.Component<
 
   componentDidCatch(error: unknown) {
     this.setState({ hasError: true, error });
+    this.props.onErrored();
```
Comment (flvndvd) [inline viz/app/components/ErrorBoundary.tsx:29]: Why not passing the error message as we use to?

**Review by flvndvd — COMMENTED** [review]
Not sure to get why we don't send the error message back to the main window?

File: front/components/assistant/conversation/actions/VisualizationActionIframe.tsx (hunk 3 above)
Comment (fontanierh) [reply front/components/assistant/conversation/actions/VisualizationActionIframe.tsx]: (explained on slack)

File: front/components/assistant/conversation/actions/VisualizationActionIframe.tsx (hunk 4 above)
Comment (fontanierh) [reply front/components/assistant/conversation/actions/VisualizationActionIframe.tsx:230]: No we need this actually

File: viz/app/components/ErrorBoundary.tsx (hunk 5 above)
Comment (fontanierh) [reply viz/app/components/ErrorBoundary.tsx:29]: We don't do anything with it yet

Comment (fontanierh) [reply viz/app/components/ErrorBoundary.tsx:29]: So we'll pass it when we need it

File: front/components/assistant/conversation/actions/VisualizationActionIframe.tsx (hunk 4 above)
Comment (flvndvd) [reply front/components/assistant/conversation/actions/VisualizationActionIframe.tsx:229]: It's already below set with tailwind, why do we need to duplicate?

Comment (fontanierh) [reply front/components/assistant/conversation/actions/VisualizationActionIframe.tsx]: Indeed, nvm me 👍

**Review by flvndvd — APPROVED** [review]

**Pull request merged**